astropy/visualization/wcsaxes/ticklabels.py

```diff
@@ -180,6 +180,10 @@ def draw(self, renderer, bboxes, ticklabels_bbox):
 
                     dist = np.hypot(dx, dy)
 
+                    # Avoid RuntimeWarning
+                    if dist == 0 or not np.isfinite(dist):
```

Comment:
Can you make a regression test for this warning so that we can investigate in more detail what the best fix is?